packages/grid/x-data-grid/src/components/panel/GridPanel.tsx

```diff
@@ -54,6 +58,8 @@ const GridPaperRoot = styled(Paper, {
 const GridPanel = React.forwardRef<HTMLDivElement, GridPanelProps>((props, ref) => {
   const { children, className, classes: classesProp, ...other } = props;
   const apiRef = useGridApiContext();
+  const rootProps = useGridRootProps();
+  const ownerState = rootProps;
```

Comment:
For `GridRoot` it's correct to pass the root props to the `ownerState`, but for any other component I think we should keep only the props received directly as owner state. We may in the future allow `GridPanel` to be used outside the grid, so the root props wouldn't be available. Could you make this change to all components, only keeping `GridRoot`?

```suggestion
  const ownerState = props;
```

Reply:
@m4theushw  Thanks for the response. I believe passing only internal props as `ownerState` is incorrect - in that case we are loosing public props of component.

As mentioned in  [`styleOverrides` docs](https://mui.com/material-ui/customization/theme-components/#overrides-based-on-props):

> The ownerState prop is a combination of public props that you pass to the component + internal state of the component.